From 65becffeb42212f8dbf9342371fa82686cb7dae0 Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Fri, 9 Mar 2012 21:38:31 +0530
Subject: [PATCH 139/172] ARM: OMAP2+: nand: check for proper return value

Return value was not checked in evm_nand_init and because of that below
crash was generated during runtime.

[    1.004381] Unable to handle kernel NULL pointer dereference at
virtual address 00000028
[    1.012864] pgd = c0004000
[    1.015688] [00000028] *pgd=00000000
[    1.019424] Internal error: Oops: 805 [#1]
[    1.023692] Modules linked in:
[    1.026881] CPU: 0    Not tainted  (3.2.0-12213-ge873350 #365)
[    1.032992] PC is at evm_nand_init+0x34/0x5c

This patch fixes the same.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 3919dab..96cfe07 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1176,6 +1176,8 @@ static void evm_nand_init(int evm_id, int profile)
 	pdata = omap_nand_init(am335x_nand_partitions,
 		ARRAY_SIZE(am335x_nand_partitions), 0, 0,
 		&am335x_nand_timings);
+	if (!pdata)
+		return;
 	pdata->ecc_opt =OMAP_ECC_BCH8_CODE_HW;
 	pdata->elm_used = true;
 	gpmc_device[0].pdata = pdata;
-- 
1.7.9.1

